packages/Search/src/DTK_CoarseGlobalSearch_def.hpp

```diff
@@ -0,0 +1,71 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_COARSEGLOBALSEARCH_DEF_HPP
+#define DTK_COARSEGLOBALSEARCH_DEF_HPP
+
+#include <Teuchos_CommHelpers.hpp>
+
+#include <Tpetra_Distributor.hpp>
+
+#include "DTK_CoarseGlobalSearch.hpp"
+#include "DTK_DetailsAlgorithms.hpp"
+
+namespace DataTransferKit
+{
+template <typename NO>
+CoarseGlobalSearch<NO>::CoarseGlobalSearch(
+    const Teuchos::RCP<const Teuchos::Comm<int>> &comm,
+    Kokkos::View<Box const *, DeviceType> local_boxes )
+    : _comm( comm )
+{
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    int comm_rank = _comm->getRank();
+    int comm_size = _comm->getSize();
+
+    _subdomain_boxes =
+        Kokkos::View<Box *, DeviceType>( "subdomain_boxes", comm_size );
+
+    // Assemble the local domain bounding box
+    int const n = local_boxes.extent( 0 );
+    Details::ExpandBoxWithBoxFunctor<DeviceType> functor( local_boxes );
+    Kokkos::parallel_reduce( "calculate_subdomain_bounding",
+                             Kokkos::RangePolicy<ExecutionSpace>( 0, n ),
+                             functor, _subdomain_boxes( comm_rank ) );
+
+    auto host_subdomain_boxes = Kokkos::create_mirror_view( _subdomain_boxes );
+    Kokkos::deep_copy( host_subdomain_boxes, _subdomain_boxes );
```

Comment:
It's not implemented by Intel MPI and I am not sure about MPICH. It is support by OpenMPI, MVAPICH, CRAY MPI, and IBM MPI. I've had by experience Intel MPI, so I try to avoid it anyway.